### DISCUSSION TITLE
プレビュー用に空の SwiftUI プロジェクトを用意する。

### DISCUSSION THREAD

**Discussion by YusukeHosonuma** [YusukeHosonuma/Effective-SwiftUI#9] [Ideas]
## Overview

Xcode には SwiftUI の View に対するプレビュー機能が搭載されており、表示確認用のプレビューを開発プロジェクトに統合した開発スタイルが実現可能となっている。

しかし、SwiftUI の View は Swift 言語で記述されているため、開発プロジェクトが大きくなるにつれてプレビュー向けへのビルドも時間がかかるようになる。

事前にローカルでプレビュー用の空のプロジェクトを用意しておき、必要なコード一式をコピペしてプレビューしながら開発すると、効率的に View のレイアウト改修作業が行える。

![image](https://user-images.githubusercontent.com/2990285/161878611-aa4ff428-d669-4273-b9f0-9dca4e6783e6.png)

また、新規 View の開発作業や、新しいコントロールの動作を確認する際にも役に立つ。

## Why not?

非常にハイスペックなマシンを利用している場合など、ビルド時間が全く気にならないレベルであれば、必要なコードをコピペする方が手間になるケースもあるかもしれない。

## Note

プロジェクトにおいてエコシステム的に利用している OSS などがあれば、それらも空のプロジェクトに導入しておくとシームレスに作業できる。

**Comment by uhooi** [comment]
Xcode より Swift Playgrounds のほうがプレビューが速い気がするので、私は後者を使うことが多いです！
